src/handle.rs

```diff
@@ -101,8 +102,16 @@ impl Handle {
 
             loop {
                 match recv(fd, buffer, length as u64, 0) {
-                    rv if rv >=0 => { nfq_handle_packet(self.ptr, buffer as *mut c_char, rv as i32); },
-                    _ => { break; }
+                    rv if rv >=0 => { 
+                        nfq_handle_packet(self.ptr, buffer as *mut c_char, length as i32);
+                    },
+                    _ => {
+                        println!("Error in reading from netlink socket. Quitting.");
+                        let e = errno();
+                        let code = e.0 as i32;
+                        println!("Error {}: {}", code, e);
```

Comment:
Use the error module in this crate, and have the fn return a result or an error or an Option<Error>.